### PR TITLE
fix directory traversal vulnerability

### PR DESCRIPTION
**Pull request by RutledgePaulV**

### CODE REVIEW

File: src/tailrecursion/ring_proxy.clj [hunk 1]
```diff
@@ -24,30 +23,41 @@
       (.read rdr buf)
       buf)))
 
+(defn join-paths
```
Comment (RutledgePaulV) [inline src/tailrecursion/ring_proxy.clj]: open to simpler constructions for this, feels clunky

**Comment by RutledgePaulV** [comment]
I've heard mention that loggly might not be in use anymore period, in which case perhaps there's value in making this a no-op piece of middleware instead and projects can remove entirely at their leisure. Is anyone able to confirm it's completely unused?

File: src/tailrecursion/ring_proxy.clj [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
   (:require
-    [clj-http.client         :refer [request]]
-    [clojure.string          :refer [join split]]
-    [ring.adapter.jetty      :refer [run-jetty]]
-    [ring.middleware.cookies :refer [wrap-cookies]] ))
+    [clj-http.client :refer [request]]
+    [clojure.string :refer [join split ends-with? starts-with? blank?]]
+    [ring.adapter.jetty :refer [run-jetty]]
+    [ring.middleware.cookies :refer [wrap-cookies]]
+    [clojure.string :as string]))
```
Comment (tessellator) [inline src/tailrecursion/ring_proxy.clj]: Can you please refer `blank?` or `:as string` on line 4 instead of including a duplicate require here?

Comment (RutledgePaulV) [reply src/tailrecursion/ring_proxy.clj]: oops, yep

**Pull request merged**